src/AutoRest.CSharp/Common/Generation/Writers/PageableMethodsWriterExtensions.cs

```diff
@@ -147,12 +147,17 @@ public static CodeWriter WritePageableBody(this CodeWriter writer, IReadOnlyList
             return writer.Line($"return {typeof(GeneratorPageableHelpers)}.{(async ? nameof(GeneratorPageableHelpers.CreateAsyncPageable) : nameof(GeneratorPageableHelpers.CreatePageable))}({createPageableParameters.Join(", ")});");
         }
 
-        private static void AddTrailingPageableParameters(this List<FormattableString> createPageableParameters, IReadOnlyCollection<Parameter> methodParameters, string scopeName, string? itemPropertyName, string? nextLinkPropertyName)
+        private static void AddTrailingPageableParameters(this List<FormattableString> createPageableParameters, IReadOnlyCollection<Parameter> methodParameters, string scopeName, string? itemPropertyName, string? nextLinkPropertyName, bool isCreateListReques = false)
         {
             createPageableParameters.Add($"{scopeName:L}");
             createPageableParameters.Add($"{itemPropertyName:L}");
             createPageableParameters.Add($"{nextLinkPropertyName:L}");
-
+            string? maxPageSize = methodParameters.Where(s => s.Name == "maxpagesize").FirstOrDefault()?.Name;
```

Comment:
put `mapagesize` to [KnowParameters ](https://github.com/Azure/autorest.csharp/blob/main/src/AutoRest.CSharp/Common/Output/Models/Shared/KnownParameters.cs)